synapse/storage/databases/main/roommember.py

```diff
@@ -862,96 +861,53 @@ async def get_mutual_rooms_between_users(
         return shared_room_ids or frozenset()
 
     async def get_joined_user_ids_from_state(
-        self, room_id: str, state: StateMap[str], state_entry: "_StateCacheEntry"
+        self, room_id: str, state: StateMap[str]
     ) -> Set[str]:
-        state_group: Union[object, int] = state_entry.state_group
-        if not state_group:
-            # If state_group is None it means it has yet to be assigned a
-            # state group, i.e. we need to make sure that calls with a state_group
-            # of None don't hit previous cached calls with a None state_group.
-            # To do this we set the state_group to a new object as object() != object()
-            state_group = object()
-
-        assert state_group is not None
-        with Measure(self._clock, "get_joined_users_from_state"):
-            return await self._get_joined_user_ids_from_context(
-                room_id, state_group, state, context=state_entry
-            )
+        """
+        For a given set of state IDs, get a map of user ID to profile information
+        for users in the room.
 
-    @cached(num_args=2, iterable=True, max_entries=100000)
-    async def _get_joined_user_ids_from_context(
-        self,
-        room_id: str,
-        state_group: Union[object, int],
-        current_state_ids: StateMap[str],
-        event: Optional[EventBase] = None,
-        context: Optional["_StateCacheEntry"] = None,
-    ) -> Set[str]:
-        # We don't use `state_group`, it's there so that we can cache based
-        # on it. However, it's important that it's never None, since two current_states
-        # with a state_group of None are likely to be different.
-        assert state_group is not None
+        This method doesn't do any fetching of data itself and instead checks various
+        caches for the relevant data before passing any remaining missing event IDs to
+        `_get_joined_profiles_from_event_ids` which does the actual data fetching.
```

Comment:
I'm somewhat confused by what this text means. Possibly not helped by the fact that `_get_joined_profiles_from_event_ids` no longer exists, but in any case, I think it's misleading:

> This method doesn't do any fetching of data itself

... except that it does. Admittedly, under the hood, it uses `_get_user_ids_from_membership_event_ids` to do that work, but that's an implementation detail. It's like saying "This method doesn't do any fetching of data itself, instead it uses `simple_select_many` to do the lookup". It might be true in some sense, but it's just not helpful information.

I think what you mean is something like  "This method checks the local event cache, before calling `_get_user_ids_from_membership_event_ids` for any uncached events." ?